### PR TITLE
Adding Pronouns to Application

### PR DESCRIPTION
**Pull request by ankykong**

### CODE REVIEW

**Comment by subhankar-panda** [comment]
Hey, looks like your last commit broke a bunch of stuff - 227 edited files 🤣 

can you revert your last commit and check your feature?

**Comment by subhankar-panda** [comment]
also whys the feature called Decaf pronouns lol?

File: src/client/pages/UserPage/components/UserProfile.tsx [hunk 1]
```diff
@@ -41,6 +42,20 @@ class UserProfile extends React.Component<Props> {
     );
   };
 
+  pronounSelect: React.StatelessComponent<CustomFieldProps> = ({ input, className }) => {
+    const pronouns = [
```
Comment (RedbackThomson) [inline src/client/pages/UserPage/components/UserProfile.tsx]: Lets re-use the values from `src/shared/UserEnums.ts` to avoid duplication errors.

**Comment by RedbackThomson** [comment]
@lisalluo Are we actually going to start using the pronouns in our messages to the applicants? If we are then this implementation will become a headache for the code.
For instance, if we want to render the same block of text considering the pronouns it would have to look something like this:
```
if(user.pronouns == 'He/Him/His') {
   message = "..."
} else if(user.pronouns == 'She/Her/Hers') {
   message = "..."
} else {
   message = "..."
}
```

While it would be better to wrap those conditional into helper functions (eg. `isUserHePronoun(user)`), it's still a little rough.
It might be better (up to you guys to decide, it adds complexity) to have a pronoun object that is populated by the fields depending on the value of the form.
```
user.pronouns = {
   subjectPronoun: "He",
   objectPronoun: "Him",
   possessivePronoun: "His"
}
```
This allows for easier templating:
```
message = `We have 3 of ${possessivePronoun} t-shirts ready.`
```

If you don't plan on using this for anything related to the website (only for badges), then I understand this is unnecessary complexity. However if there is a possibility of in the future switching to using it, then it would be better to reconsider the design from the beginning.

**Comment by lisalluo** [comment]
> @lisalluo Are we actually going to start using the pronouns in our messages to the applicants? If we are then this implementation will become a headache for the code.
> For instance, if we want to render the same block of text considering the pronouns it would have to look something like this:
> 
> ```
> if(user.pronouns == 'He/Him/His') {
>    message = "..."
> } else if(user.pronouns == 'She/Her/Hers') {
>    message = "..."
> } else {
>    message = "..."
> }
> ```
> 
> While it would be better to wrap those conditional into helper functions (eg. `isUserHePronoun(user)`), it's still a little rough.
> It might be better (up to you guys to decide, it adds complexity) to have a pronoun object that is populated by the fields depending on the value of the form.
> 
> ```
> user.pronouns = {
>    subjectPronoun: "He",
>    objectPronoun: "Him",
>    possessivePronoun: "His"
> }
> ```
> 
> This allows for easier templating:
> 
> ```
> message = `We have 3 of ${possessivePronoun} t-shirts ready.`
> ```
> 
> If you don't plan on using this for anything related to the website (only for badges), then I understand this is unnecessary complexity. However if there is a possibility of in the future switching to using it, then it would be better to reconsider the design from the beginning.

Right now, pronouns are just going to be for name tags. We don't have any plans to do any messaging that uses users' pronouns

**Comment by lisalluo** [comment]
@ankykong looks like you missed a couple "Pronoun" -> "Pronouns".
- on the application page
- on the admin "view all users" page
I think that's all of them but double check!

File: src/server/models/User.ts [hunk 2]
```diff
@@ -63,6 +63,13 @@ export const UserSchema = new Schema({
     public: true,
     editable: true,
   },
+  // Declares the user's pronoun
+  pronoun: {
```
Comment (RedbackThomson) [inline src/server/models/User.ts]: `pronouns`?

File: src/shared/api/Requests.ts [hunk 3]
```diff
@@ -87,6 +87,7 @@ export interface RegisterUserPersonalSectionRequest {
   lastName: string;
   birthdate: string;
   gender: string;
+  pronoun?: string;
```
Comment (RedbackThomson) [inline src/shared/api/Requests.ts]: `pronouns`?

**Pull request merged**